src/core/entity.factory.ts

```diff
@@ -11,6 +13,11 @@ export class EntityFactory extends Repository {
     }
     return thread;
   }
+
+  public directThreadFromResponse(responseItem: DirectInboxFeedResponseThreadsItem): DirectThreadEntity {
```

Comment:
This method is definitely violates the single responsibility principle for the `EntityFactory`. We need to refactor it